.github/workflows/java_tests.yml

```diff
@@ -0,0 +1,160 @@
+    name: "Check GCP variables set"
+    runs-on: ubuntu-latest
+    outputs:
+      gcp-variables-set: ${{ steps.check_gcp_variables.outputs.gcp-variables-set }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Check are GCP variables set"
+        run: "./scripts/ci/ci_check_are_gcp_variables_set.sh"
+        id: check_gcp_variables
+        env:
+          GCP_PROJECT_ID: ${{ secrets.GCP_PROJECT_ID }}
+          GCP_SA_EMAIL: ${{ secrets.GCP_SA_EMAIL }}
+          GCP_SA_KEY: ${{ secrets.GCP_SA_KEY }}
+          GCP_TESTING_BUCKET: ${{ secrets.GCP_TESTING_BUCKET }}
+
+  java_unit_tests:
+    name: 'Java Unit Tests'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      # :sdks:java:core:test
+      - name: Run :sdks:java:core:test
+        run: ./gradlew :sdks:java:core:test
+      - name: Upload test logs for :sdks:java:core:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-sdks-java-core-test-${{ matrix.os }}
+          path: sdks/java/core/build/reports/tests/test/**
+      # :sdks:java:harness:test
+      - name: Run :sdks:java:harness:test
+        if: ${{ always() }}
+        run: ./gradlew :sdks:java:harness:test
+      - name: Upload test logs for :sdks:java:harness:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-sdks-java-harness-test-${{ matrix.os }}
+          path: sdks/java/harness/build/reports/tests/test/**
+      # :runners:core-java:test
+      - name: Run :runners:core-java:test
+        if: ${{ always() }}
+        run: ./gradlew :runners:core-java:test
+      - name: Upload test logs for :runners:core-java:test
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_unit_tests-runners-core-java-test-${{ matrix.os }}
+          path: runners/core-java/build/reports/tests/test/**
+
+  java_wordcount_direct_runner:
+    name: 'Java Wordcount Direct Runner'
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run WordCount Unix
+        shell: bash
+        run: |
+          bash ./gradlew -p examples/ \
+             integrationTest -DintegrationTestPipelineOptions='[
+          	 "--runner=DirectRunner", "--tempRoot=./tmp" ]' \
+             --tests org.apache.beam.examples.WordCountIT \
+             -DintegrationTestRunner=direct
+      - name: Upload test logs
+        uses: actions/upload-artifact@v2
+        if: always()
+        with:
+          name: java_wordcount_direct_runner-${{matrix.os}}
+          path: examples/java/build/reports/tests/integrationTest
```

Comment:
Why upload the whole directory here instead of a globstar? Just curious.

Reply:
I am not sure did I understand. Is question about why there is `examples/java/build/reports/tests/integrationTest` instead of `examples/java/build/reports/tests/integrationTest/**`?
I it was a question, I will simplify paths for `java_unit_tests` logs paths to avoid unnecessary globstar.

Reply:
Yes, this was the question. If there is no reason to differentiate, a consistent path is preferred. Thanks!